From 5d42d8ff071e808c361b1529c43a2c34a741a271 Mon Sep 17 00:00:00 2001
From: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date: Tue, 19 Apr 2022 21:01:14 +0300
Subject: [PATCH 110/157] net: ethernet: renesas: rswitch: fix issue with
 interrupts for unused chains

In some cases we might get interrupt request for an unused chain. We
just need to ignore it or otherwise next logic will break.

Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

Moved check upper to completely avoid access to unused chain.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>

diff --git a/drivers/net/ethernet/renesas/rswitch.c b/drivers/net/ethernet/renesas/rswitch.c
index 09ecb7d501bf..eedbe1f35487 100644
--- a/drivers/net/ethernet/renesas/rswitch.c
+++ b/drivers/net/ethernet/renesas/rswitch.c
@@ -3124,6 +3124,8 @@ static irqreturn_t __maybe_unused rswitch_data_irq(struct rswitch_private *priv,
 	int index, bit;
 
 	for (i = 0; i < priv->gwca.num_chains; i++) {
+		if (!test_bit(i, priv->gwca.used))
+			continue;
 		c = &priv->gwca.chains[i];
 		index = c->index / 32;
 		bit = BIT(c->index % 32);
-- 
2.25.1

